DeepLinkSDK/RouteHandler/DPLRouteHandler.h

```diff
@@ -51,6 +58,10 @@
  */
 - (UIViewController *)viewControllerForPresentingDeepLink:(DPLDeepLink *)deepLink;
 
+/**
+ returns the first `UINavigationController'
+ */
+- (UINavigationController *)navigationControllerForPresentingDeepLink:(DPLDeepLink *)deepLink;
```

Comment:
`DPLRouteHandler` is intended to be subclassed and `-viewControllerForPresentingDeepLink:` overridden. The interface is intentionally generic to allow you to return whatever view controller you like. 

The work you've done in `-navigationControllerForPresentingDeepLink:` could be done in `-viewControllerForPresentingDeepLink:` in your own subclass of `DPLRouteHandler`.